tests/integration/helpers.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helpers for cephfs-client integration tests."""
+
+import json
+import logging
+from time import sleep
+from typing import Any, Tuple
+
+from charms.storage_libs.v0.cephfs_interfaces import CephFSAuthInfo, CephFSShareInfo
+from pylxd import Client
+
+_logger = logging.getLogger(__name__)
+
+FS_NAME = "cephfs"
+USERNAME = "fs-client"
+PATH = "/"
+
+
+def bootstrap_microceph() -> Tuple[CephFSShareInfo, CephFSAuthInfo]:
+    """Bootstrap a minimal Microceph cluster in LXD.
+
+    Returns: A tuple with the share info and auth info to mount the CephFS share.
+    """
+    client = Client()
+    if client.instances.exists("microceph"):
+        _logger.info("Microceph server already exists")
+        instance = client.instances.get("microceph")
+        return _get_cephfs_info(instance)
+
+    _logger.info("Bootstrapping Microceph cluster")
+    config = {
+        "name": "microceph",
+        "source": {
+            "alias": "jammy/amd64",
+            "mode": "pull",
+            "protocol": "simplestreams",
+            "server": "https://cloud-images.ubuntu.com/releases",
+            "type": "image",
+        },
+        "type": "virtual-machine",
+    }
+    client.instances.create(config, wait=True)
+    instance = client.instances.get(config["name"])
+    instance.start(wait=True)
+    _logger.info("Installing Microceph inside LXD container")
+
+    time = 2
+    for x in range(0, 10):
+        try:
+            _exec_command(instance, ["echo", "'Hello!'"])
+            err = None
+        except Exception as e:
+            err = e
+
+        if err:
+            _logger.info(f"Sleeping for {time} seconds")
+            sleep(time)
+            time *= 2
+        else:
+            break
+
+    if err:
+        raise err
```

Comment:
We can use [tenacity](https://tenacity.readthedocs.io/en/latest/) here to continuously test if the LXD agent has been started on the VM we have provisioned.